source/blog/2018-07-19-ovn-dpdk.html.md

```diff
@@ -0,0 +1,90 @@
+
+For additional information, refer to the role's [repository](https://github.com/ovirt/ovirt-ansible-dpdk-setup/).
+
+### OVN Integration
+DPDK in oVirt now leverages the capabilities of OVN in the form of an OVN localnet. This allows for seamless connectivity across OVN networks, benefiting from OVN's software defined routers, switches, security groups, and ACL's.
+
+For more information about OVN localnet integration in oVirt, refer to oVirt's [Provider Physical Network RFE](https://ovirt.org/develop/release-management/features/network/provider-physical-network/).
+
+# Usage in oVirt
+1) Install `oVirt.dpdk-setup` Ansible role via ansible-galaxy:
+    ```
+    # ansible-galaxy install oVirt.dpdk-setup
+    ```
+2) Execute the role as described above.
+3) Unless the host was already configured, the host will restart for kernel changes to be applied. Note that after reboot the following values are changed (If they are not, you may have incompatible hardware, or you should report a bug on ovirt@users.org):
+    * IOMMU: ```/sys/devices/virtual/iommu/<DMAR device>/devices/<PCI address>``` should exist. DMAR devices are typically annotated by dmar0, dmar1, and so forth.
+    * hugepages: ```grep Huge /proc/meminfo``` should reflect the desired state as defined in the Ansible playbook (i.e. `Hugepagesize` and `HugePages_Total`)
+    * CPU partitioning: based on the devices that are to be used with a DPDK compatible driver, CPUs should be partitioned separately between each NUMA node. Refer to `lscpu` to see live CPU NUMA separation information, e.g.: `NUMA node1 CPU(s):     8-15`
+
+    Note: NICs using userspace drivers do not appear in the kernel; `ip` commands won't list devices using userspace drivers. in oVirt, such devices appear as `dpdk0`, `dpdk1`, and so forth.
+4) Create an oVirt network ("phys-net-demo") on your data center and attach it to your cluster.
+![](https://www.ovirt.org/images/ovn-dpdk/create_phys_net.png)
+5) Attach "phys-net-demo" to DPDK NICs accross your cluster.
+![](https://www.ovirt.org/images/ovn-dpdk/set_phys_net.png)
+6) Create an external OVN network ("ext-net-demo"), setting "phys-net-demo" as its physical network. This configuration between "ext-net-demo" and "phys-net-demo" will enable traffic between "phys-net-demo" (which DPDK's physical port is part of) and the rest of the ports present in OVN's "ext-net-demo" network.
+![](https://www.ovirt.org/images/ovn-dpdk/create_ext_net.png)
+7) "ext-net-demo" vNic's may now be added to virtual machines. These vNics now share L2 connectivity with other remote ports in "ext-net-demo" via DPDK's NIC (and other local ports internally).
+![](https://www.ovirt.org/images/ovn-dpdk/ext_net_demo.png)
+8) Temporarily, SElinux needs to be set to permissivie. This is due to an [open bug](https://bugzilla.redhat.com/1598435) where SElinux blocks the creation of a QEMU socket. Once the bug is resolved, this step should be skipped.
```

Comment:
Please correct "permissivie" to permissive
Maybe add that to perform a "one the spot" change use:
setenforce 0
But that will only last until the next reboot
or edit /etc/selinux/config for a permanent change since the kernel boots using this config file